manager/encryption/encryption.go

```diff
@@ -0,0 +1,132 @@
+package encryption
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/docker/swarmkit/api"
+	"github.com/gogo/protobuf/proto"
+	"github.com/pkg/errors"
+)
+
+// This package defines the interfaces and encryption package
+
+const humanReadablePrefix = "SWMKEY-1-"
+
+// ErrCannotDecode is the type of error returned when some data cannot be decoded as plaintext
+type ErrCannotDecode struct {
```

Comment:
I'm not sure if we should introduce this error type if we aren't switching on it. Just use `errors.Errorf`.